dace/transformation/interstate/loop_to_map.py

```diff
@@ -291,6 +291,48 @@ def test_read_memlet(self, sdfg: SDFG, state: SDFGState, edge: gr.MultiConnector
             return False
 
         return True
+    
+    def _is_array_thread_local(self, name: str, itervar: str, sdfg: SDFG, states: List[SDFGState]) -> bool:
+        """
+        This helper method checks whether an array used exclusively in the body of a detected for-loop is thread-local,
+        i.e., its whole range is may be used in every loop iteration, or is can be shared by multiple iterations.
+
+        For simplicity, it is assumed that the for-loop can be safely transformed to a Map. The method applies only to
+        bodies that become a NestedSDFG.
+
+        :param name: The name of array.
+        :param itervar: The for-loop iteration variable.
+        :param sdfg: The SDFG containing the states that comprise the body of the for-loop.
+        :param states: A list of states that comprise the body of the for-loop.
+        :return: True if the array is thread-local, otherwise False.
+        """
+
+        desc = sdfg.arrays[name]
+        if not isinstance(desc, dt.Array):
+            # Scalars are always thread-local.
```

Comment:
That will not work for streams, please change condition

Reply:
Do you think that it is safe to apply this for Streams?